libs/schemas/src/queries/user.schemas.ts

```diff
@@ -1,43 +1,84 @@
+import { Roles } from '@hicommonwealth/shared';
 import { z } from 'zod';
-import { Community } from '../entities';
+import { Community, Thread } from '../entities';
 import { Comment } from '../entities/comment.schemas';
 import { Tags } from '../entities/tag.schemas';
-import { Thread } from '../entities/thread.schemas';
 import { Address, UserProfile } from '../entities/user.schemas';
+import { PG_INT } from '../utils';
+import { PaginatedResultSchema, PaginationParamsSchema } from './pagination';
 
-export const GetNewProfileReq = z.object({
-  userId: z.string().optional(),
-});
+export const GetUserProfile = {
+  input: z.object({
+    userId: PG_INT.optional(),
+  }),
+  output: z.object({
+    userId: PG_INT,
+    profile: UserProfile,
+    totalUpvotes: z.number().int(),
+    addresses: z.array(
+      Address.extend({
+        Community: Community.pick({
+          id: true,
+          base: true,
+          ss58_prefix: true,
+        }),
+      }),
+    ),
+    threads: z.array(Thread),
+    comments: z.array(
+      Comment.extend({
+        Thread: z.undefined(),
+        community_id: z.string(),
+      }),
+    ),
+    commentThreads: z.array(Thread),
+    isOwner: z.boolean(),
+    tags: z.array(Tags.extend({ id: PG_INT })),
+  }),
+};
 
-export const GetNewProfileResp = z.object({
-  userId: z.number(),
-  profile: UserProfile,
-  totalUpvotes: z.number().int(),
+export const SearchUserProfilesView = z.object({
+  user_id: PG_INT,
+  profile_name: z.string(),
+  avatar_url: z.string().nullish(),
+  created_at: z.date().or(z.string()),
+  last_active: z.date().or(z.string()).nullish(),
   addresses: z.array(
-    Address.extend({
-      Community: Community.pick({
-        id: true,
-        base: true,
-        ss58_prefix: true,
-      }),
+    z.object({
+      id: PG_INT,
+      community_id: z.string(),
+      address: z.string(),
+      role: z.enum(Roles),
     }),
   ),
-  threads: z.array(Thread),
-  comments: z.array(Comment),
-  commentThreads: z.array(Thread),
-  isOwner: z.boolean(),
-  tags: z.array(Tags),
+  //group_ids: z.array(PG_INT),
```

Comment:
what about it?

Reply:
forgot to remove this, we don't use it anywhere